### PR TITLE
Add an icon to annotations when they have comments

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-11429
affects: Student, Teacher
release note: Added an icon to annotations on submissions when there are comments attached

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/o0w88ay) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fo0w88ay)
teacher: [Link](https://rebrand.ly/4ranc7l) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F4ranc7l)
parent: [Link](https://rebrand.ly/g4142zu) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fg4142zu)

**Review by inst-danger — COMMENTED** [review]
fix lint

File: Core/Core/DocViewer/DocViewerCommentReplyAnnotation.swift [hunk 1]
```diff
@@ -21,3 +21,52 @@ import PSPDFKit
...
+}
+
+class DocViewerStrikeOutAnnotation: StrikeOutAnnotation {
+    open override func draw(context: CGContext, options: RenderOptions?) {
+        super.draw(context: context, options: options)
+        drawRepliesIcon(in: context, options: options)
+    }
+}
+
+
```
Comment (inst-danger) [inline Core/Core/DocViewer/DocViewerCommentReplyAnnotation.swift:52]:
```suggestion
```

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/DocViewer/DocViewerCommentReplyAnnotation.swift:52:1: Vertical Whitespace Violation: Limit vertical whitespace to a single empty line. Currently 2. (vertical_whitespace)
    [student]
    [teacher]
</details>


[MBL-11429](https://instructure.atlassian.net/browse/MBL-11429)

#### Affected Apps: Student, Teacher

#### Release Note: 
Added an icon to annotations on submissions when there are comments attached
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/DocViewer/DocViewerAnnotationProvider.swift [hunk 2]
```diff
@@ -44,10 +44,15 @@ class DocViewerAnnotationProvider: PDFContainerAnnotationProvider {
         super.init(documentProvider: documentProvider)
 
         guard metadata.enabled else { return }
+        var hasReplies: Set<String> = []
```
[the remark this answers is not in the document]
Comment (vanwagonet) [reply Core/Core/DocViewer/DocViewerAnnotationProvider.swift:47]: I named it after the property they were going to populate, but I suppose haveReplies would have been appropriate too.

**Pull request merged**